File: Cpp/Client-dessin/Polygon.cpp
#include "Polygon.h"
#include "FormVisitor.h"

polygon::Polygon::Polygon(string id, vector<Vecteur2D*> listPoint, string color) : Form(id, color)
{
	copy(listPoint.begin(), listPoint.end(), back_inserter(_pointList));
}

polygon::Polygon::operator string() const
{
	string out = "type:polygon,id:" + _id + ",coord:";
	for (Vecteur2D* point : _pointList)
	{
		out += string(*point);
	}
	out += ",color:" + this->Form::_color;

	return out;
}

void polygon::Polygon::accept(const FormVisitor* formVisitor)
{
	formVisitor->visite(this);
}

File: Java/Client-Server/src/form/Polygon.java
package form;

import java.awt.Graphics;
import java.util.ArrayList;

public class Polygon extends Form {
	/**
	 *  Cree un polygone du nombre de point fourni
	 */
	
	private String color;
	
	public Polygon(String id, ArrayList<Vecteur2D> pointList, String color) {
		super(id);
		for(int i = 0; i < pointList.size(); i++) {
			this.pointList.add(pointList.get(i));
		}
		this.setColor(color);
		
	}
	
	/**
	 * Dessine le polygone en récupérant tous les abscisses et toutes les ordonnees
	 */
	@Override
	public void draw(Graphics g) {
		int n = pointList.size();
		int xTab[] = new int[n];
		int yTab[] = new int[n];
		for(int i = 0; i < pointList.size(); i++) {
			int temp[] = pointList.get(i).arrondi();
			xTab[i] = temp[0];
			yTab[i] = temp[1];
		}
		g.setColor(getColor(color));
		g.fillPolygon(xTab, yTab, n);
	}

	public String getColor() {
		return color;
	}

	public void setColor(String color) {
		this.color = color;
	}

}
